typescript/infra/src/utils/metrics.ts

```diff
@@ -35,3 +36,25 @@ export async function submitMetrics(
     `Prometheus metrics pushed to PushGateway with status ${statusCode}`,
   );
 }
+
+/**
+ * Start a simple HTTP server to host metrics. This just takes the registry and dumps the text
+ * string to people who request `GET /metrics`.
+ *
+ * PROMETHEUS_PORT env var is used to determine what port ot host on, defaults to 9090.
+ */
+export function startMetricsServer(register: Registry): http.Server {
+  return http
+    .createServer((req, res) => {
+      if (req.url != 'metrics') res.writeHead(404, 'Invalid url').end();
+      if (req.method != 'GET') res.writeHead(405, 'Invalid method').end();
+
+      register
+        .metrics()
+        .then((metricsStr) => {
+          res.writeHead(200, { ContentType: 'text/plain' }).end(metricsStr);
+        })
+        .catch((err) => console.error(err));
```

Comment:
why not `async/await`?

Reply:
Because node's create server does not expect an async fn so the work to wrap it would be a lot more than just doing it like this.

Reply:
makes sense though this particular snippet looks like it will lead to unexpected behavior